### PR TITLE
BHoM_Engine: Update IElement references with the Dimensional_oM

### LINKED ISSUE #1488: Update IElement references with the Dimensional_oM
`IElement` interfaces are migrating to `Dimensional_oM`

This will cause every project which uses any of the `IElement`s or any class inheriting from `IElement` to need a reference to `Dimensional_oM`. 
Further more, any file using any of the `IElement` interfaces explicitly, will need to add 
`using BH.oM.Dimensional;`

### Branch Name:
`Dimensional_oM698-Migrate-IElementXD-Interfaces-from-Geometry_oM`

##### For every project in your toolkit
1. If you had a reference to Geometry_oM, add a reference to Dimensional_oM

##### For every file in your toolkit
2. If you have any direct use of any of the `IElement` interfaces, add `using BH.oM.Dimensional;`

### PR DESCRIPTION
**Pull request by kThorsager**
### NOTE: Depends on 

https://github.com/BHoM/BHoM/pull/704

### Issues addressed by this PR

Closes #1488 
Adds references to Dimensional_oM where necessary and 'using' statments where that is necessary to comply with the changes.

And since we're messing in every project regardless, I added the Spatial_Engine's dependencies

### Changelog
- Add dependencies for Spatial_oM

### CODE REVIEW

File: Architecture_Engine/Architecture_Engine.csproj [hunk 1]
```diff
@@ -38,6 +38,10 @@
       <HintPath>..\..\BHoM\Build\BHoM.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Dimensional_oM, Version=3.0.0.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (FraserGreenroyd) [inline Architecture_Engine/Architecture_Engine.csproj]: Reference should not contain version, culture, or processorArchitecture.
Reference should also have `copy local` set to false.

File: Common_Engine/Common_Engine.csproj [hunk 2]
```diff
@@ -39,6 +39,10 @@
       <HintPath>..\..\BHoM\Build\Common_oM.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Dimensional_oM, Version=3.0.0.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (FraserGreenroyd) [inline Common_Engine/Common_Engine.csproj]: Reference should not contain version, culture, or processorArchitecture.
Reference should also have `copy local` set to false.

File: Environment_Engine/Environment_Engine.csproj [hunk 3]
```diff
@@ -43,6 +43,10 @@
       <HintPath>..\..\BHoM\Build\BHoM.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Dimensional_oM, Version=3.0.0.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (FraserGreenroyd) [inline Environment_Engine/Environment_Engine.csproj]: Reference should not contain version, culture, or processorArchitecture.
Reference should also have `copy local` set to false.

File: Geometry_Engine/Geometry_Engine.csproj [hunk 4]
```diff
@@ -34,6 +34,10 @@
       <HintPath>..\..\BHoM\Build\BHoM.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Dimensional_oM, Version=3.0.0.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (FraserGreenroyd) [inline Geometry_Engine/Geometry_Engine.csproj]: Reference should not contain version, culture, or processorArchitecture.
Reference should also have `copy local` set to false.

File: Humans_Engine/Humans_Engine.csproj [hunk 5]
```diff
@@ -50,6 +50,10 @@
       <HintPath>..\..\BHoM\Build\BHoM.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Dimensional_oM, Version=3.0.0.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (FraserGreenroyd) [inline Humans_Engine/Humans_Engine.csproj]: Reference should not contain version, culture, or processorArchitecture.
Reference should also have `copy local` set to false.

File: Physical_Engine/Physical_Engine.csproj [hunk 6]
```diff
@@ -35,6 +35,10 @@
       <HintPath>..\..\BHoM\Build\BHoM.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Dimensional_oM, Version=3.0.0.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (FraserGreenroyd) [inline Physical_Engine/Physical_Engine.csproj]: Reference should not contain version, culture, or processorArchitecture.
Reference should also have `copy local` set to false.

File: Structure_Engine/Structure_Engine.csproj [hunk 7]
```diff
@@ -49,6 +49,10 @@
       <HintPath>..\..\BHoM\Build\Data_oM.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Dimensional_oM, Version=3.0.0.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (FraserGreenroyd) [inline Structure_Engine/Structure_Engine.csproj]: Reference should not contain version, culture, or processorArchitecture.
Reference should also have `copy local` set to false.

**Comment by kThorsager** [comment]
Given that that is a general compliance thing I took the liberty to remove version and processor architecture from some other BHoM references as well.

**Comment by FraserGreenroyd** [comment]
> Given that that is a general compliance thing I took the liberty to remove version and processor architecture from some other BHoM references as well.

That's the kind of forward thinking I can get behind :wink: thanks @kThorsager - will review ASAP

**Pull request merged**